returnn/torch/engine.py

```diff
@@ -514,7 +523,10 @@ def _load_optimizer(self, epoch):
         :param int epoch: Epoch from which to load the optimizer state.
         """
         filename = self.get_epoch_model_filename(epoch=epoch - 1) + ".opt.pt"
-        self._updater.load_optimizer(filename, device=self._device)
+        if (os.path.isfile(filename)):
```

Comment:
maybe we also introduce a flag for this, so you do have to explicitly allow this? I am not sure about the implications, resetting the optimizer states definitely leads to a different training run.